File: Java/23. Merge k Sorted Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        List<Integer> list = new ArrayList<>();
        for (int i = 0; i < lists.length; i++) {
            while(lists[i] != null){
                list.add(lists[i].val);
                lists[i] = lists[i].next;
            }
        }
        Collections.sort(list);
        ListNode temp = new ListNode();
        ListNode ans = temp;
        for (int i: list){
            temp.next = new ListNode(i);
            temp=temp.next;
        }
        return ans.next;
    }
}

File: C#/23. Merge k Sorted Lists.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MergeKLists(ListNode[] lists) {
        List<int> list = new List<int>();
        for (int i = 0; i < lists.Length; i++) {
            while(lists[i] != null){
                list.Add(lists[i].val);
                lists[i] = lists[i].next;
            }
        }
        list.Sort();
        ListNode temp = new ListNode();
        ListNode ans = temp;
        foreach (int i in list){
            temp.next = new ListNode(i);
            temp=temp.next;
        }
        return ans.next;
    }
}
